### ISSUE TITLE
RP Lidar S2 ERROR (Linux & Windows)

### ISSUE THREAD

**Issue by AnOrdinaryUsser** [Slamtec/rplidar_ros#107]
I have recently acquired a lidar S2, when I put the rplidar_ros package in my workspace and compile, everything goes fine. But when I try to run some of the launches of the lidar S2, like:
- view_rplidar_s2.launch
- rplidar_s2.launch

I get the following error:
```bash
[ERROR] [1679570485.482521809]: Error, rplidar internal error detected. Please reboot the device to r
[rplidarNode-2] process has died [pid 16358, exit code 255, cmd 
```

I am working on the following platform:
- Jetson Nano (ARM)
- Ubuntu 18.04
- ROS Melodic

As I didn't get good results on linux because I couldn't solve the previous error, I went on to see if I could use the software provided by slamtec (roboStudio).
In order to use this software I need to download in theory the Slamware Frame Grabber plugin, however I get the following error and this prevents me from using the software to view my S2 lidar.

![image](https://user-images.githubusercontent.com/61872281/227199177-47f30c1f-5dfa-4fc3-89f6-e1a466cad970.png)

**Comment by AnOrdinaryUsser** [comment]
**Some updates:**

I have been in contact with SLAMTEC to see if they could offer me a solution for using this type of lidar. They proposed this type of solution to me:

a) Check whether the USB power supply cable and USB communication cable are plugged in firmly and whether the USB port is normal, replace the USB port to verify.

b) Check whether the USB power supply is stable. It is not recommended for customers to use the USB hub for power supply. The quality of the USB hub cannot be guaranteed.

c) Directly connect the two USBs of the lidar to the computer at the same time (within 3 seconds), and use the studio to check whether the data is normal. When restarting, be sure to plug and unplug the two USBs (within 3 seconds) at the same time to restart.

d) Use a 5V 2A mobile phone charger to power the radar separately. Use the official website RS to check whether the data is normal. When restarting, be sure to plug and unplug the two USBs (within 3 seconds) at the same time to restart.

**_What I have tried and it has worked for me:_**

1) If I connect my JETSON NANO to a charger direct to the light it gives me on the 5V 4A output and the lidar directly connected to the JETSON ports it works correctly and stabilized.

2) If I connect my JETSON NANO to a 5V 4A converter which gives me 5V 4A through a battery that reduces its voltage from 36V to 5V, the lidar only works for 3 seconds.

3) If you connect your JETSON NANO to a laboratory power supply, the same happens as I mentioned above in the second option.

4) Another option is to power the S2 lidar through a 5V 2A charger, which has a USB, but I have also had no success if the JETSON is not powered through a power socket.  

SLAMTEC has provided me with a manual or a brief summary of how to set up the S2 lidar, but I am still having the same problems. But in case anyone needs it, it is as follows: https://files.fm/u/m27uxtrpv 

I have also been provided with a roboStudio in which you can test lidars, and there the S2 lidar works correctly. I also leave the link here: https://files.fm/u/n5tw238cn

**Comment by deyouslamtec** [comment]
@AnOrdinaryUsser Have you solved your problem?

**Comment by AnOrdinaryUsser** [comment]
> deyouslamtec

@deyouslamtec In the end, after everything I put in and tested, the Jetson Nano reached its maximum capacity and could not supply enough power for itself and the lidar, so it would have to be powered externally.
The problem is that this lidar presents a rather complicated connection problem and that is that most of the time the two usb cables, both the data and the power cables, must be connected at the same time.
I was developing a closed prototype in which I don't always have access to the cables, so when I started it would almost always work, but if I stopped it I would have to turn the robot completely off and start again. This was very frustrating and I decided not to use this lidar and use an A2 or A3 lidar instead of the A1 I was using.
